layers/+completion/spacemacs-ivy/packages.el

```diff
@@ -265,6 +300,26 @@ Helm hack."
         (call-interactively 'describe-mode))
 
       (define-key counsel-find-file-map (kbd "C-h") 'counsel-up-directory)
+
+      ;; set additional ivy actions
+      (ivy-set-actions
+       'counsel-find-file
+       spacemacs-ivy-file-actions)
+
+      (defun spacemacs-ivy/find-file-vsplit (file)
+        "find file in vertical split"
+        (interactive)
+        (select-window (split-window-right))
+        (find-file file)
```

Comment:
Calling `select-window` and then `find-file` assumes that `find-file` always uses the selected window, but it may choose another window if the user customizes (or advises) `find-file` or `switch-to-buffer`. Maybe there's a `find-file*` variant that takes an `action` argument similar to `pop-to-buffer`? If I have time later I will offer an alternative implementation.